src/tests/gov/nasa/jpf/test/vm/reflection/ProxyTest.java

```diff
@@ -134,13 +134,19 @@ public void run() {
 
   @Test
   public void testProxyCreationInCaseOfChoiceGenerator() {
-    if (verifyNoPropertyViolation()){
+    if (verifyNoPropertyViolation()) {
+      MyHandler handler = new MyHandler(42);
+      Ifc ifc = (Ifc) Proxy.newProxyInstance(Ifc.class.getClassLoader(), new Class[] { Ifc.class }, handler);
+      String interfaceName = Ifc.class.getName();
+      String packageName = interfaceName.substring(0, interfaceName.lastIndexOf('.'));
+      String desiredProxyClsName = packageName + ".$Proxy$"
+              + Integer.toHexString(Ifc.class.getName().hashCode());
+      assertEquals(ifc.getClass().getName(), desiredProxyClsName);
+
+      // now test cross-thread behavior with minimal operations
+      // which mean we need fewer interleavings
       NewThread t = new NewThread();
       t.start();
-      MyHandler handler = new MyHandler(42);
```

Comment:
If you move all code of the current thread up to before the new thread is created, there is nothing left to interleave for JPF. It may be the case that this test cannot be optimized without losing the fact that it verifies thread interleavings here.